module-2/backend/week_three.md

```diff
@@ -10,16 +10,39 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What are the 3 main features in an ERD?
+  + Entities
+  + Relations
+  + Cardinalities
 2. Create an ERD for the following database: Restaurants, Customers, Items, Ingredients, Beverages, Orders, Vendors.
 3. What is the entry at the command line to create a new rails app?
+  + `rails new my_cool_app`
 4. What do Models generally inherit from in rails?
+  + In Rails 4 and earlier, `ActiveRecord::Base`. In Rails 5, `ApplicationRecord`
 5. What do Controllers generally inherit from in a rails project?
+  + `ApplicationController`
 6. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  + `resources :horses, only: [:show]`
 7. What rake task is useful when looking at routes, and what information does it give you?
+  + `rake routes` gives you the prefix, the verb, the path, and the controller action.
 8. What is an example of a route helper? When would you use them?
+  + `horse_path(id)` is a helper rails would give you to access the horse route created in question 6. It would be usually be used in a controller or view, but it could be used anywhere that you want to link to something.
 9. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+  + `_url` returns the absolute path and `_path` returns the relative path.
 10. What are strong params and why are the necessary?
+  + Strong params are params that are limited to only accept certain input. They are necessary to prevent unwanted things from being sent in to the application.
 11. What role does `form_for` play in helping us create our forms?
+  + `form_for` determines where a form is and, based on that, where to submit that form.
 12. How does `form_for` know where to submit the user's input?
+  + It contextually knows based on what view page the form is on (so if it is on new, it will submit to create)
```

Comment:
it knows based on the object we pass it as an argument, not necessarily what view the form is in 👍